apps/dav/lib/Connector/Sabre/File.php

```
@@ -453,9 +453,15 @@ private function createFileChunked($data) {
 					/** @var \OC\Files\Storage\Storage $targetStorage */
 					list($partStorage, $partInternalPath) = $this->fileView->resolvePath($partFile);
 
-
 					$chunk_handler->file_assemble($partStorage, $partInternalPath);
 
+					//assembly might take a while, double check db connection is still alive
+					$conn = \OC::$server->getDatabaseConnection();
+					if ($conn->ping() === false) {
+						$conn->close();
```

are we in a transaction here? This will result in inconsistent state ...

---

I'd throw an exception in here - we are in an unclear state. The request has to be performed again.